Content.Client/UserInterface/Systems/Radial/Controls/RadialContainer.xaml.cs

```diff
@@ -0,0 +1,403 @@
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using Content.Client.Message;
+using Robust.Client.Animations;
+using Robust.Client.AutoGenerated;
+using Robust.Client.Graphics;
+using Robust.Client.UserInterface;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Animations;
+
+namespace Content.Client.UserInterface.Systems.Radial.Controls;
+
+[GenerateTypedNameReferences, Virtual]
+public partial class RadialContainer : Control
+{
+    private EntityUid? _attachedEntity;
+
+    private bool _isOpened = false;
+    private const int MaxButtons = 8;
+
+    private const string MoveAnimationKey = "move";
+    private const string InSizeAnimationKey = "insize";
+    private const string OutSizeAnimationKey = "outsize";
+
+    /// <summary>
+    /// Fired, when radial was died(closed)
+    /// </summary>
+    public Action<RadialContainer>? OnClose;
+
+    public Action<RadialContainer>? OnAttached;
+    public Action<RadialContainer>? OnDetached;
+
+    /// <summary>
+    /// Radial item size, when cursor was focused on button
+    /// </summary>
+    public float FocusSize { get; set; } = 64f;
+
+    /// <summary>
+    /// Normal radial item size, when cursor not focused
+    /// </summary>
+    public float NormalSize { get; set; } = 50f;
+
+    /// <summary>
+    /// Items moving animation time, when radial was opened
+    /// </summary>
+    public float MoveAnimationTime { get; set; } = 0.3f;
+
+    /// <summary>
+    /// Item focus sizing animation, when cursor focused and leaved
+    /// </summary>
+    public float FocusAnimationTime { get; set; } = 0.25f;
+
+    /// <summary>
+    /// Show or not item action text on radial menu
+    /// </summary>
+    public bool ActionVisible { get; set; } = true;
+
+    /// <summary>
+    /// Used for UIController! Radial menu position offset
+    /// </summary>
+    public float VerticalOffset { get; set; } = 0.0f;
+
+    /// <summary>
+    /// Used for UIController! Radius, where is container was available
+    /// </summary>
+    public float VisionRadius { get; set; } = 10.0f;
```

Comment:
```suggestion
    /// <summary>
    /// Radial item size when the cursor hovers the button.
    /// </summary>
    public float FocusSize = 64f;

    /// <summary>
    /// Normal radial item size when not hovered.
    /// </summary>
    public float NormalSize = 50f;

    /// <summary>
    /// Animation time for child RadialItems to get their final position.
    /// </summary>
    public float MoveAnimationTime = 0.3f;

    /// <summary>
    /// Item hover sizing animation on entry and exit.
    /// </summary>
    public float FocusAnimationTime = 0.25f;

    /// <summary>
    /// Show action text on radial menu
    /// </summary>
    public bool ActionVisible = true;

    /// <summary>
    /// Used for UIController! Radial menu position offset
    /// </summary>
    public float VerticalOffset = 0.0f;

    /// <summary>
    /// Used for UIController! Radius, where is container was available
    /// </summary>
    public float VisionRadius = 10.0f;
```

Also rename all instances of focus to hover to match engine. Vertifal offset also needs to specify it's in world terms.